lib/maru.ex

```diff
@@ -57,8 +57,12 @@ defmodule Maru do
     end
   end
 
+  json_library =  Application.get_env(:maru, :json_library, Jason)
```

Comment:
I think that the note in the docs is enough. This would emit a warning at compile time of `maru`, which means that it'll happen only the first time someone fetches and compiles the dependencies. Most likely they wouldn't have configured the `:json_library` at this point 🙂

Reply:
Hmmm, I haven't got your point, do you think there's no need to check json library here?

Reply:
Yes. This check will be performed only when `maru` is compiled. Dependencies of applications are only compiled once, unless they are changed or someone explicitly executes `mix deps.compile <app>`.

This means that the warning will be shown only when the user fetches and compiles `maru` dependency for the first time in their app. But most probably, at the moment of fetching the dep, they wouldn't have configured the `:json_library` yet. Which means that everyone will see the warning even though they haven't done anything wrong.

Reply:
but if people follow the README, `jason` will be added to `mix.exs`, and by default the `json_library` is `Jason`, so I think people won't see any error.

Reply:
but there's a known issue, when people want to change json library, the whole files in `_build` folder have to be deleted and compile again 🙂

Reply:
Yes, in this case it would happen. But if you revert again to `Application.get_env(:maru, :json_library, Jason)` everything should work correctly without recompilation 🙂 Anyway, these are only my suggestions - I'm happy that `:json_library` got to the README! 🎉

Reply:
maru is really diffrent with ecto, people can use ecto without json library, but maru is heavy depends on json, so I think we should make it a compile error, because the application can't work without a json library.
Thanks 🙂